can/interfaces/vector/canlib.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+"""
+Ctypes wrapper module for Vector CAN Interface on win32/win64 systems
+Authors: Julien Grave <grave.jul@gmail.com>, Christian Sandberg
+"""
+# Import Standard Python Modules
+# ==============================
+import ctypes
+import logging
+import sys
+import time
+
+# Import Modules
+# ==============
+from can import BusABC, Message
+
+# Define Module Logger
+# ====================
+LOG = logging.getLogger('can.vector')
+
+# Import safely Vector API module for Travis tests
+if sys.platform == 'win32':
+    from . import vxlapi
+else:
+    LOG.warning('Vector API does not work on %s platform', sys.platform)
+    vxlapi = None
+
+
+class VectorBus(BusABC):
+    """The CAN Bus implemented for the Vector interface."""
+
+    def __init__(self, channel, can_filters=None, poll_interval=0.005,
+                 app_name="CANalyzer", **config):
+        """
+        :param list channel:
+            The channel indexes to create this bus with.
+            Can also be a single integer or a comma separated string.
+        :param float poll_interval:
+            Poll interval in seconds.
+        :param str app_name:
+            Name of application in Hardware Config.
+        """
+        if vxlapi is None:
+            raise ImportError("The Vector API has not been loaded")
+        self.poll_interval = poll_interval
+        if isinstance(channel, (list, tuple)):
+            self.channel = channel
+        elif isinstance(channel, int):
+            self.channel = [channel]
+        else:
+            # Assume comma separated string of channels
+            self.channel = [int(channel) for channel in channel.split(',')]
+        self._app_name = app_name.encode()
+        self._driver_init(can_filters)
+        try:
+            self._go_on_bus()
+        except vxlapi.VectorError:
+            self.shutdown()
+            raise
+        super(VectorBus, self).__init__()
+
+    def _driver_init(self, can_filters):
+        xl_status = vxlapi.open_driver()
+        LOG.debug('Open Driver: Status: %d', xl_status.value)
+        if xl_status.value == vxlapi.XL_SUCCESS:
+            channel_mask = []
+            # Get channels masks
+            for channel in self.channel:
+                app_channel = ctypes.c_uint(channel)
+                hw_type = ctypes.c_uint(0)
+                hw_index = ctypes.c_uint(0)
+                hw_channel = ctypes.c_uint(0)
+                xl_status = vxlapi.get_appl_config(
+                    app_name=self._app_name,
+                    app_channel=app_channel,
+                    hw_type=hw_type,
+                    hw_index=hw_index,
+                    hw_channel=hw_channel)
+                if xl_status.value == vxlapi.XL_SUCCESS:
```

Comment:
I forgot to remove this check since an exception will be raised if it is not successful anyway.